src/server/server.ts

```diff
@@ -176,6 +176,50 @@ namespace ts.server {
 
     // Override sys.write because fs.writeSync is not reliable on Node 4
     ts.sys.write = (s: string) => writeMessage(s);
+    
+    const serverHost: ServerHost = <ServerHost>ts.sys;
+    serverHost.globalTypingCachePath = toPath(
+        ".typingCache",
+        process.env[(process.platform === "win32") ? "USERPROFILE" : "HOME"],
+        createGetCanonicalFileName(sys.useCaseSensitiveFileNames));
+
+    let tsd: any;
+    serverHost.getTsd = () => {
+        if (tsd !== undefined) {
+            return tsd;
+        }
+        try {
+            tsd = require("tsd");
+            return tsd;
+        }
+        catch (e) {
+            logger.msg("Failed to require tsd.");
+        }
+        return undefined;
+    };
+    serverHost.cachedTsdJsons = {};
+    serverHost.getTsdJson = (tsdJsonPath: Path) => {

```

Comment:
I can't find where in the guideline we say to avoid type annotations. Wouldn't type annotation in this case is better so you know what is the type of input parameter?